File: EASY/69. Sqrt/sqrt.py
# Given a non-negative integer x, return the square root of x rounded down to the nearest integer.
# The returned integer should be non-negative as well.

# You must not use any built-in exponent function or operator.
# For example, do not use pow(x, 0.5) in c++ or x ** 0.5 in python.

# Accepted

class Solution:
    def mySqrt(self, x: int) -> int:
        l, r = 0, x + 1
        while l < r:
            mid = (l + r) // 2
            if mid * mid > x:
                r = mid
            else:
                l = mid + 1
        return l - 1


s = Solution()
print(s.mySqrt(4))

File: EASY/69. Sqrt/Sqrt.cpp
//
// Created by Savin Vladimir on 26.10.2022.
//

// Accepted

#include <iostream>

class Solution {
public:
    int mySqrt(int x) {
        long long int l = 1, r = x;
        while (l <= r) {
            long long int mid = l + (r - l) / 2;
            if (mid * mid == x)
                return mid;
            else if (mid * mid > x)
                r = mid - 1;
            else l = mid + 1;
        }
        return r;
    }
};

int main() {

    Solution obj;
    std::cout << obj.mySqrt(9);

    return 0;
}

